tests/contrib/hooks/test_gcp_transfer_hook.py

```diff
@@ -129,14 +283,347 @@ def test_wait_for_transfer_job(self, mock_sleep):
         )
         mock_sleep.assert_called_once_with(TIME_TO_SLEEP_IN_SECONDS)
 
-    def test_wait_for_transfer_job_failed(self):
-        mock_list = self.conn.transferOperations.return_value.list
-        mock_execute = mock_list.return_value.execute
-        mock_execute.side_effect = [
-            {'operations': [{'name': 'test-job', 'metadata': {'status': 'FAILED'}}]},
-        ]
+    @mock.patch('time.sleep')
+    @mock.patch('airflow.contrib.hooks.gcp_transfer_hook.'
+                'GCPTransferServiceHook.get_conn')
+    def test_wait_for_transfer_job_failed(self, get_conn, mock_sleep):
+        list_method = get_conn.return_value.transferOperations.return_value.list
+        list_execute_method = list_method.return_value.execute
+        list_execute_method.return_value = \
+            {
+                'operations': [{
+                    'name': TRANSFER_OPERATION_NAME,
+                    'metadata': {
+                        'status': GcpTransferOperationStatus.FAILED
+                    }
+                }]
+            }
+
+        get_conn.return_value.transferOperations \
+            .return_value.list_next \
+            .return_value = None
+
         with self.assertRaises(AirflowException):
-            self.transfer_hook.wait_for_transfer_job({
+            self.gct_hook.wait_for_transfer_job({
                 'projectId': 'test-project',
                 'name': 'transferJobs/test-job',
             })
+            self.assertTrue(list_method.called)
+
+    @mock.patch('time.sleep')
+    @mock.patch('airflow.contrib.hooks.gcp_transfer_hook.'
+                'GCPTransferServiceHook.get_conn')
+    def test_wait_for_transfer_job_expect_failed(self, get_conn, mock_sleep):
+        list_method = get_conn.return_value.transferOperations.return_value.list
+        list_execute_method = list_method.return_value.execute
+        list_execute_method.return_value = {
+            'operations': [{
+                'name': TRANSFER_OPERATION_NAME,
+                'metadata': {
+                    'status': GcpTransferOperationStatus.FAILED}}
+            ]
+        }
+
+        get_conn.return_value.transferOperations \
+            .return_value.list_next \
+            .return_value = None
+
+        self.gct_hook.wait_for_transfer_job(
+            job={
+                'projectId': 'test-project',
+                'name': 'transferJobs/test-job',
+            },
+            expected_status=GcpTransferOperationStatus.FAILED
+        )
+        self.assertTrue(True)
```

Comment:
Why this :) ? Not needed.

Reply:
Haha cracked me up! 😂

Reply:
I will fix it. Some debug stuff. I am happy that I could give you happiness. Your happiness is my happiness

Reply:
Fixed